client/src/components/AddBookModal/AddBookModal.tsx

```diff
@@ -39,22 +41,30 @@ const AddBookModal: React.FC<AddBookModalProps> = ({closeModalFunc, addBookFunct
   // }, [])
```

Comment:
Looking a bit more at the code, I'm guessing you've started off with it as a single object so that you can just pass it into `addBookFunc` in one go, but with this approach that should still be pretty trivial:

```jsx
onClick={() => addBookFunc({ date, review, availableToBorrow, ...otherStuff })}
```